l10n_ch_base_bank/models/bank.py

```diff
@@ -236,34 +251,53 @@ def _check_postal_num(self):
         return True
 
     @api.multi
-    def _get_acc_name(self):
-        """ Return an account name for a bank account
-        to use with a ccp for ISR.
-        This method makes sure to generate a unique name
-        """
-
+    def _update_acc_name(self):
+        """Check if number generated from ccp, if yes replace it on new """
         part_name = self.partner_id.name
         if not part_name and self.env.context.get('default_partner_id'):
             partner_id = self.env.context.get('default_partner_id')
             part_name = self.env['res.partner'].browse(partner_id)[0].name
+        if part_name or 'CCP' in self.acc_number:
+            self.acc_number = self._compute_name_ccp(part_name, self.ccp)
+        elif not part_name:
+            self.acc_number = _("Undefined")
 
-        if part_name:
-            acc_name = _("Bank/CCP {}").format(part_name)
+    @api.multi
+    def _compute_name_ccp(self, partner_name, ccp):
+        """This method makes sure to generate a unique name"""
+        if partner_name:
+            acc_name = _("{}/CCP-{}").format(partner_name, ccp)
```

Comment:
Please remove the dash. CCP is not part of the number.
```suggestion
            acc_name = _("{}/CCP {}").format(partner_name, ccp)
```